pkg/reconciler/cache/cachedresourceendpointslice/cachedresourceendpointslice_controller.go

```diff
@@ -70,32 +68,32 @@ func NewController(
-			return indexers.ByIndex[*cachev1alpha1.CachedResourceEndpointSlice](
-				cachedResourceEndpointSliceClusterInformer.Informer().GetIndexer(),
-				byCachedResourceAndLogicalCluster,
-				cachedResourceEndpointSliceByCachedResourceAndLogicalCluster(
-					&cachev1alpha1.CachedResourceReference{
-						Name: cachedResource.Name,
-					},
-					logicalcluster.From(cachedResource),
-				),
-			)
+		listCachedResourceEndpointSlices: func() ([]*cachev1alpha1.CachedResourceEndpointSlice, error) {
+			return cachedResourceEndpointSliceClusterInformer.Lister().List(labels.Everything())
 		},
-		getCachedResourceEndpointSlice: func(clusterName logicalcluster.Name, name string) (*cachev1alpha1.CachedResourceEndpointSlice, error) {
-			return cachedResourceEndpointSliceClusterInformer.Cluster(clusterName).Lister().Get(name)
+		getCachedResourceEndpointSlice: func(path logicalcluster.Path, name string) (*cachev1alpha1.CachedResourceEndpointSlice, error) {
+			return indexers.ByPathAndName[*cachev1alpha1.CachedResourceEndpointSlice](cachev1alpha1.Resource("cachedresourceendpointslice"), cachedResourceEndpointSliceClusterInformer.Informer().GetIndexer(), path, name)
 		},
-		getCachedResource: func(clusterName logicalcluster.Name, name string) (*cachev1alpha1.CachedResource, error) {
-			return cachedResourceClusterInformer.Cluster(clusterName).Lister().Get(name)
+		getCachedResource: func(path logicalcluster.Path, name string) (*cachev1alpha1.CachedResource, error) {
+			return indexers.ByPathAndName[*cachev1alpha1.CachedResource](cachev1alpha1.Resource("cachedresource"), globalCachedResourceClusterInformer.Informer().GetIndexer(), path, name)
 		},
-		getMyShard: func() (*corev1alpha1.Shard, error) {
-			return globalShardClusterInformer.Cluster(core.RootCluster).Lister().Get(shardName)
+		getPartition: func(clusterName logicalcluster.Name, name string) (*topologyv1alpha1.Partition, error) {
+			return partitionClusterInformer.Lister().Cluster(clusterName).Get(name)
 		},
-		getLogicalCluster: func(clusterName logicalcluster.Name) (*corev1alpha1.LogicalCluster, error) {
-			return lcClusterInformer.Cluster(clusterName).Lister().Get("cluster")
-		},
-		getAPIBinding: func(clusterName logicalcluster.Name, bindingName string) (*apisv1alpha2.APIBinding, error) {
-			return apiBindingClusterInformer.Cluster(clusterName).Lister().Get(bindingName)
+		getCachedResourceEndpointSlicesByPartition: func(key string) ([]*cachev1alpha1.CachedResourceEndpointSlice, error) {
+			list, err := cachedResourceEndpointSliceClusterInformer.Informer().GetIndexer().ByIndex(indexCachedResourceEndpointSlicesByPartition, key)
+			if err != nil {
+				return nil, err
+			}
+			var slices []*cachev1alpha1.CachedResourceEndpointSlice
+			for _, obj := range list {
+				slice, ok := obj.(*cachev1alpha1.CachedResourceEndpointSlice)
```

Comment:
index already has this check. Do we need to have it here too? 
If index returned something else - we have bigger problem? no?